spyder/plugins/console.py

```diff
@@ -101,12 +101,6 @@ def __init__(self, parent=None, namespace=None, commands=[], message=None,
         self.dismiss_error = False
 
     #------ Private API --------------------------------------------------------
-    def set_historylog(self, historylog):
-        """Bind historylog instance to this console
-        Not used anymore since v2.0"""
-        historylog.add_history(self.shell.history_filename)
-        self.shell.append_to_history.connect(historylog.append_to_history)
-

```

Comment:
This needs to stay. It's used to save and load history in the internal console. So please put it back it.